### PR TITLE
Overhaul coverage setup

### LINKED ISSUE #433: Test-drive grcov for coverage reports
@marco-c advises `grcov` over `llvm-cov` for coverage reports. Test-drive it. Compare and contrast.

### PR DESCRIPTION
**Pull request by mvines**
This PR gets coverage working again.  Humans may run `./scripts/coverage.sh` locally to generate a report.  

CI runs `./scripts/coverage.sh` on PRs and Pushes if any .rs files are affected.  Results are attached to the buildkite job and uploaded to codecov.io.

`cargo-cov` usage has been removed, it's overly complex and unnecessary.

Note that in the buildkite job for this PR, there's a message at the top telling saying "Coverage skipped as no .rs files were modified" -- we skip coverage if the PR has no .rs file changes to help with CI load.  Contrast that with the demo PR at #2211.

Fixes #433

### CODE REVIEW

**Comment by mvines** [comment]
This all works well on macOS, but blows up on Linux still.  🐧

**Comment by mvines** [comment]
The problem on linux appears to be that the gnco files from the different crates all end up getting written to the same `build.gcno`/`lib.gcno` file in the root of the workspace wheres on macOS each crate gets it's own unique set of gcno`s.   Looks like it's time to download the cargo-cov source and figure out what's going on

**Comment by mvines** [comment]
`cargo-cov` appears to be a dead end and even somewhat harmful.  Getting really close to coverage working with plain old cargo nestled in a cozy, yet light, blanket of bash.

File: scripts/coverage.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+rm -rf target/cov/$reportName
+
+_ cargo +nightly build --target-dir target/cov --all
+_ cargo +nightly test --target-dir target/cov --lib --all
+
+_ scripts/fetch-grcov.sh
+echo "--- grcov"
+./grcov target/cov/debug/deps/ > target/cov/lcov_full.info
+
+echo "--- filter_non_local_files_from_lcov"
```
Comment (marco-c) [inline scripts/coverage.sh]: You should be able to use the `-s` option of grcov to achieve the same result (ignore any file which is not in the repository)

Comment (mvines) [reply scripts/coverage.sh]: Oh sweet, thanks.

Comment (mvines) [reply scripts/coverage.sh]: 💔 
```bash
$ ./grcov target/cov/debug/deps/ -s . > out.lcov
thread 'main' panicked at 'Only one file in the repository should end with src/packet.rs (sdk/src/packet.rs and src/packet.rs both end with that)', src/path_rewriting.rs:143:13
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Comment (marco-c) [reply scripts/coverage.sh]: I'll fix this.

Comment (garious) [reply scripts/coverage.sh]: Hey @marco-c, what's generally the best way to contact you?  Care to join our Discord forum? https://discord.gg/k8Kq58

**Pull request merged**